src/ucp/core/ucp_ep.h

```diff
@@ -192,6 +192,17 @@ enum {
 };
 
 
+/**
+ * Endpoint configuration comparison flags
+ */
+enum {
+    /* Compare lanes layout */
+    UCP_EP_CONFIG_CMP_FLAG_LANES = UCS_BIT(0),
+    /* Strong compare of all fields */
+    UCP_EP_CONFIG_CMP_MASK_ALL   = UCS_MASK(8)
+};
```

Comment:
why needed? can use boolean instead

Reply:
1) just for readability
2) can be easily extended

Reply:
IMO simplicity is prefferd here

Reply:
well, what's simpler?
```
if (ucp_ep_config_is_equal(key1, key2,
                           UCP_EP_CONFIG_CMP_FLAG_LANES)) {
    ...
}
```
or
```
if (ucp_ep_config_is_equal(key1, key2, 1)) {
    ...
}
```
and now mix it with
`ucp_ep_config_is_equal(key1, key2, 0)` or `ucp_ep_config_is_equal(key1, key2, UCP_EP_CONFIG_CMP_MASK_ALL)`
IMO internal func implementation is secondary vs reading of common logic of wireup

Reply:
for readability it's possible to define a const and pass to function IMO, but anyway for your decision.

Reply:
but even if enum remains, why need UCS_BIT/UCS_MASK? can use simple values

Reply:
because `UCP_EP_CONFIG_CMP_FLAG_LANES` is logical subset of `UCP_EP_CONFIG_CMP_MASK_ALL`

Reply:
IMO there's no need for bitmask, because the only comparison is in line 1988

Reply:
it's used widely on caller's side for readability and consistency

Reply:
will be equally readable with simple values rahter than bitmask

Reply:
I'm not agree, set of flags is mask, `UCP_EP_CONFIG_CMP_FLAG_LANES` is subset of `UCP_EP_CONFIG_CMP_MASK_ALL`

Reply:
@evgeny-leksikov @shasson5 
IMO the issue is not of significance, though I slightly prefer the existing approach, mostly for the reason of clarity.